sunpy/map/maputils.py

```diff
@@ -80,7 +80,22 @@ def map_edges(smap):
     return top, bottom, left_hand_side, right_hand_side
 
 
-@u.quantity_input
+def _is_coordinate_helioprojective(coordinates):
+    """
+    Checks if the coordinate is Helioprojective
+
+    Parameters
+    ----------
+    coordinates : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frame`
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` if coordinates are Helioprojective, `False` otherwise.
+    """
+    return isinstance(coordinates.frame, Helioprojective)
```

Comment:
```suggestion
    frame = coordinates.frame if hasattr(coordinates, 'frame') else coordinates
    if not isinstance(frame, Helioprojective):
        raise ValueError("The input coordinate(s) must be in the Helioprojective frame.")
```
The first line is important to allow for both `SkyCoord` instances and frame instances.